test/src/main/java/org/apache/accumulo/test/IteratorEnvIT.java

```diff
@@ -71,121 +92,123 @@ public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoo
    */
-  public static class MajcIter extends WrappingIterator {
+  public static class MajcIter extends Filter {
     IteratorScope scope = IteratorScope.majc;
 
     @Override
     public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options,
         IteratorEnvironment env) throws IOException {
       super.init(source, options, env);
-      testEnv(scope, options, env);
-      try {
-        env.isUserCompaction();
-      } catch (IllegalStateException e) {
-        throw new RuntimeException("Test failed");
-      }
-      try {
-        env.isFullMajorCompaction();
-      } catch (IllegalStateException e) {
-        throw new RuntimeException("Test failed");
-      }
+      // Checking for compaction on majc should not throw an error.
+      testEnv(scope, options, env, false);
+    }
+
+    @Override
+    public boolean accept(Key k, Value v) {
+      // The only reason for filtering out some data is as a way to verify init() and testEnv()
+      // have been called
+      return !k.getColumnFamily().toString().equals(BAD_COL_FAM);
     }
   }
 
   /**
    *
    */
-  public static class MincIter extends WrappingIterator {
+  public static class MincIter extends Filter {
     IteratorScope scope = IteratorScope.minc;
 
     @Override
     public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options,
         IteratorEnvironment env) throws IOException {
       super.init(source, options, env);
-      testEnv(scope, options, env);
-      try {
-        env.isUserCompaction();
-        throw new RuntimeException(
-            "Test failed - Expected to throw IllegalStateException when checking compaction on a scan.");
-      } catch (IllegalStateException e) {}
-      try {
-        env.isFullMajorCompaction();
-        throw new RuntimeException(
-            "Test failed - Expected to throw IllegalStateException when checking compaction on a scan.");
-      } catch (IllegalStateException e) {}
+      // Checking for compaction on minc should throw an error.
+      testEnv(scope, options, env, true);
+    }
+
+    @Override
+    public boolean accept(Key k, Value v) {
+      // The only reason for filtering out some data is as a way to verify init() and testEnv()
+      // have been called
+      return !k.getColumnFamily().toString().equals(BAD_COL_FAM);
     }
   }
 
   /**
    * Test the environment methods return what is expected.
    */
-  private static void testEnv(IteratorScope scope, Map<String,String> opts,
-      IteratorEnvironment env) {
-    TableId expectedTableId = TableId.of(opts.get("expected.table.id"));
+  private static void testEnv(IteratorScope scope, Map<String,String> opts, IteratorEnvironment env,
+      boolean checkCompactionShouldThrow) {
+    String expTableIdString = opts.get(EXPECTED_TABLE_ID_OPT);
+    TableId expectedTableId = expTableIdString == null ? null : TableId.of(expTableIdString);
+
+    if (checkCompactionShouldThrow) {
```

Comment:
Could use the scope instead of adding the new boolean.

```suggestion
    if (scope != majc) {
```